src/main/scala/org/downtowndailybread/bethsaida/request/AttendanceRequest.scala

```diff
@@ -15,40 +15,62 @@ class AttendanceRequest(val settings: Settings, val conn: Connection)
     with UUIDProvider {
 
   def getAttendanceById(attendanceId: UUID): Attendance = {
-    val sql = getAttendanceSql("a.id = ?")
+    val sql = getAttendanceSql("id = ?")
     val ps = conn.prepareStatement(sql)
     ps.setString(1, attendanceId)
 
     getSingle(ps.executeQuery(), createRs)
   }
 
   def getAttendanceByClientId(clientId: UUID): Seq[Attendance] = {
-    val sql = getAttendanceSql("a.client_id = ?")
+    val sql = getAttendanceSql("client_id = ?")
     val ps = conn.prepareStatement(sql)
     ps.setString(1, clientId)
 
     createSeq(ps.executeQuery(), createRs)
   }
 
   def getAttendanceByEventId(eventId: UUID): Seq[Attendance] = {
-    val sql = getAttendanceSql("a.event_id = ?")
+    val sql = getAttendanceSql("event_id = ?")
     val ps = conn.prepareStatement(sql)
     ps.setString(1, eventId)
 
     createSeq(ps.executeQuery(), createRs)
   }
 
-  def updateAttendance(attendanceId: UUID, attrib: AttendanceAttribute)(
+  def updateAttendance(attendanceId: UUID, attendance: Attendance)(
     implicit user: InternalUser
   ): Unit = {
-    insertAttendanceAttributeInternal(attendanceId, attrib, true)
+    val sql =
+      s"""
+         |update attendance
+         |set check_in_time  = ?,
+         |    check_out_time = ?,
+         |    event_id       = cast(? as uuid),
+         |    client_id      = cast(? as uuid)
+         |from attendance
+         |where id = cast(? as uuid)
+       """.stripMargin
+    val ps = conn.prepareStatement(sql)
+    ps.setZonedDateTime(1, attendance.attribute.checkInTime)
+    ps.setZonedDateTime(2, attendance.attribute.checkInTime)
```

Comment:
Should both of these times be the same?

Reply:
Yeah, for now the schema doesn't exactly match the data model, so in the future I'll fix this. I was aware of this while writing it.